src/main/cli/process.ts

```diff
@@ -0,0 +1,133 @@
+import * as glob from "glob";
+import * as path from "path";
+import { _PACKAGING } from "readium-desktop/preprocessor-directives";
+import * as yargs from "yargs";
+
+import { cli_, cliImport, cliOpds, cliRead } from "./commandLine";
+
+declare const __APP_NAME__: string;
+declare const __APP_VERSION__: string;
+
+// Logger
+const debug = debug_("readium-desktop:cli");
+
+export function cli(mainFct: () => void) {
+
+    yargs
+        .scriptName(__APP_NAME__)
+        .version(__APP_VERSION__)
+        .usage("$0 <cmd> [args]")
+        .command("opds <title> <url>",
+            "import opds feed",
+            (y) =>
+                y.positional("title", {
+                    describe: "title opds feed",
+                    type: "string",
+                })
+                .positional("url", {
+                    describe: "url opds feed",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                const promise = cliOpds(argv.title, argv.url);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("import <path>",
+            "import epub or lpcl file",
+            (y) =>
+                y.positional("path", {
+                    describe: "absolute, relative or globbing path",
+                    type: "string",
+                })
+                .example("import", "\"./myPath/**/*.epub\"")
+                .example("import", "\"/*.epub\"")
+                .example("import", "\"myPublication.epub\"")
+            ,
+            (argv) => {
+                const pathArray = glob.sync(argv.path, {
+                    absolute: true,
+                    realpath: true,
+                }) || [];
+                const promise = cliImport(pathArray);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("read <title>",
+            "searches already-imported publications with the provided TITLE, and opens the reader with the first match",
+            (y) =>
+                y.positional("title", {
+                    describe: "title of your publication",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                mainFct();
+                app.on("ready", async () => {
+                    try {
+                        if (!await cliRead(argv.title)) {
+                            debug("no publication to read");
```

Comment:
Should there be a `dialog.showErrorBox()` here as well?

Reply:
done